test/test_unistd.rs

```diff
@@ -457,3 +457,82 @@ fn test_canceling_alarm() {
     assert_eq!(alarm::set(60), None);
     assert_eq!(alarm::cancel(), Some(60));
 }
+
+#[test]
+fn test_getpwnam() {
+    let res = User::query( UserQuery::Name("root".to_string()) ).unwrap();
+    assert!(res.unwrap().uid == Uid::from_raw(0));
```

Comment:
This isn't a very good test.  That field would've been zero even if `getpwnam` did nothing, because you initialize the struct to zero.  Can you find a more interesting field to check?

Reply:
I transformed this into a doctest. About to push it.